client/object_put.go

```diff
@@ -0,0 +1,199 @@
+	v2session "github.com/nspcc-dev/neofs-api-go/v2/session"
+	"github.com/nspcc-dev/neofs-sdk-go/object"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"github.com/nspcc-dev/neofs-sdk-go/session"
+	"github.com/nspcc-dev/neofs-sdk-go/token"
+)
+
+// PrmObjectPutInit groups parameters of ObjectPutInit operation.
+//
+// At the moment the operation is not parameterized, however,
+// the structure is still declared for backward compatibility.
+type PrmObjectPutInit struct{}
+
+// ResObjectPut groups the final result values of ObjectPutInit operation.
+type ResObjectPut struct {
+	statusRes
+
+	resp v2object.PutResponse
+}
+
+// ReadStoredObjectID reads identifier of the saved object.
+// Returns false if ID is missing (not read).
+func (x *ResObjectPut) ReadStoredObjectID(id *oid.ID) bool {
+	idv2 := x.resp.GetBody().GetObjectID()
+	if idv2 == nil {
+		return false
+	}
+
+	*id = *oid.NewIDFromV2(idv2) // need smth better
+
+	return true
+}
+
+// ObjectWriter is designed to write one object to NeoFS system.
+//
+// Must be initialized using Client.ObjectPutInit, any other
+// usage is unsafe.
+type ObjectWriter struct {
+	cancelCtxStream context.CancelFunc
+
+	ctxCall contextCall
+
+	// initially bound tp contextCall
+	metaHdr v2session.RequestMetaHeader
+
+	// initially bound to contextCall
+	partInit v2object.PutObjectPartInit
+
+	chunkCalled bool
+
+	partChunk v2object.PutObjectPartChunk
+}
+
+// UseKey specifies private key to sign the requests.
+// If key is not provided, then Client default key is used.
+func (x *ObjectWriter) UseKey(key ecdsa.PrivateKey) {
+	x.ctxCall.key = key
+}
+
+// WithBearerToken attaches bearer token to be used for the operation.
+// Should be called once before any writing steps.
+func (x *ObjectWriter) WithBearerToken(t token.BearerToken) {
+	x.metaHdr.SetBearerToken(t.ToV2())
+}
+
+// WithinSession specifies session within which object should be stored.
+// Should be called once before any writing steps.
+func (x *ObjectWriter) WithinSession(t session.Token) {
+	x.metaHdr.SetSessionToken(t.ToV2())
+}
+
+// MarkLocal tells the server to execute the operation locally.
+func (x *ObjectWriter) MarkLocal() {
+	x.metaHdr.SetTTL(1)
+}
+
+// WriteHeader writes header of the object. Result means success.
+// Failure reason can be received via Close.
+func (x *ObjectWriter) WriteHeader(hdr object.Object) bool {
+	v2Hdr := hdr.ToV2()
+
+	x.partInit.SetObjectID(v2Hdr.GetObjectID())
+	x.partInit.SetHeader(v2Hdr.GetHeader())
+	x.partInit.SetSignature(v2Hdr.GetSignature())
+
+	return x.ctxCall.writeRequest()
+}
+
+// WritePayloadChunk writes chunk of the object payload. Result means success.
+// Failure reason can be received via Close.
+func (x *ObjectWriter) WritePayloadChunk(chunk []byte) bool {
+	if !x.chunkCalled {
+		x.chunkCalled = true
+		x.ctxCall.req.(*v2object.PutRequest).GetBody().SetObjectPart(&x.partChunk)
+	}
+
+	for ln := len(chunk); ln > 0; ln = len(chunk) {
+		if ln > 512 {
```

Comment:
how about constant with some comments? (i do not know what 512 means)

Reply:
512B?

Reply:
and what is the meaning of that 512B?

Reply:
That we send payload in 512B chunks.